### ISSUE TITLE
Propagate locale from CarbonPeriod to iterated dates

### ISSUE THREAD

**Issue by Arany** [briannesbitt/Carbon#1443]
Hello,

The locale is not forwarded to created dates.
```php
$period  = CarbonPeriod::since('2018-03-01')
    ->locale('de')
    ->month()
    ->until('2018-04-01');

foreach ($period as $date) {
     echo $date->isoFormat('MMMM YYYY');
}
```
Output:

Carbon version: 2.2.0
PHP version: 7.1.20

I expected to get:
```
März 2018
April 2018
```

But I actually get:
```
March 2018
April 2018
```

I can get the correct results by adding ``locale`` inside the foreach loop:
```php
$period  = CarbonPeriod::since('2018-03-01')
    ->month()
    ->until('2018-04-01');

foreach ($period as $date) {
    echo $date->locale('de')->isoFormat('MMMM YYYY');
}
```

Thanks!

**Comment by kylekatarnls** [comment]
Merged, will be available on 2.3 tag release.

**Comment by kylekatarnls** [comment]
2.3 released
